lib/cryptoexchange/exchanges/txbit/services/market.rb

```diff
@@ -0,0 +1,50 @@
+module Cryptoexchange::Exchanges
+  module Txbit
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            true
+          end
+        end
+
+        def fetch(market_pair)
+          output = super(ticker_url(market_pair))
+          adapt(output, market_pair)
+        end
+
+        def ticker_url(market_pair)
+          base = market_pair.base
+          target = market_pair.target
+          # Txbit pair has BTC comes first, when BTC is typically a Target not a Base
+          "#{Cryptoexchange::Exchanges::Txbit::Market::API_URL}/public/getmarketsummary?market=#{target}/#{base}"
+        end
+
+        def adapt(output, market_pair)
+          handle_invalid(output)
+          ticker = Cryptoexchange::Models::Ticker.new
+          market = output['result'][0]
```

Comment:
are you reading the output correctly?

Reply:
Thank you @tmlee I was able to solve the issue. Btw, You guys have a very nice system for specs and testing.